### PR TITLE
core: add option to exclude categories from the SEO indexing (0.x)

### PR DESCRIPTION
**Pull request by mworrell**
### Description

This adds an option to exclude all pages in a category from indexing by a search engine.

For this the property `is_seo_noindex_cat` is set 

Also update the Docs GH Action to keep working with a newer Ubuntu version as GH is removing 18.04 support.

### Checklist

- [ ] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

File: modules/mod_seo_sitemap/support/seo_sitemap.erl [hunk 1]
```diff
@@ -83,7 +83,8 @@ categories(Context) ->
     Cats = z_db:q("select id from rsc where category_id = $1", [ CatCat ], Context),
     PIds = lists:filtermap(
         fun({Id}) ->
-            case z_acl:rsc_visible(Id, Context) of
+            IsSeoNoIndexCat = z_convert:to_bool(m_rsc:p_no_acl(Id, is_seo_noindex_cat, Context)),
+            case IsSeoNoIndexCat andalso z_acl:rsc_visible(Id, Context) of
```
Comment (robvandenbogaard) [inline modules/mod_seo_sitemap/support/seo_sitemap.erl:87]: Shouldn't this condition be `(not IsSeoNoIndexCat) andalso visible`?

File: modules/mod_base/controllers/controller_file.erl [hunk 2]
```diff
@@ -136,6 +139,18 @@ provide_content(ReqData,  {Info,Context} = State) ->
...
 
+is_noindex(#z_file_info{acls=Acls}, Context) ->
+    lists:any(
+        fun
+            (Id) when is_integer(Id) ->
+                CatId = m_rsc:p_no_acl(Id, category_id, Context),
+                z_convert:to_bool(m_rsc:p_no_acl(Id, seo_noindex, Context))
+                orelse z_convert:to_bool(m_rsc:p_no_acl(CatId, is_seo_noindex_cat, Context));
+            (_) ->
+                false
```
Comment (robvandenbogaard) [inline modules/mod_base/controllers/controller_file.erl:150]: Should non-resources be indexed?

File: modules/mod_seo_sitemap/models/m_seo_sitemap.erl [hunk 3]
```diff
@@ -164,7 +168,7 @@ maybe_add_category_attrs(#{ category_id := CatId } = Map, Context) ->
                 undefined -> <<"weekly">>;
                 CF -> CF
             end,
-            Map1#{ changefreq := Freq };
+            Map1#{ changefreq => Freq };
```
Comment (robvandenbogaard) [inline modules/mod_seo_sitemap/models/m_seo_sitemap.erl:171]: Good side-catch 😬

File: modules/mod_admin/templates/_admin_edit_content_seo.tpl [hunk 4]
```diff
@@ -87,24 +87,54 @@
...
+                <div class="col-md-9 col-md-offset-3">
+                    <div class="checkbox">
+                        <label>
+                            <input id="seo_noindex"
+                                type="checkbox"
+                                name="seo_noindex"
+                                value="1"
+                                {% if not is_editable %}disabled{% endif %}
+                                {% if id.seo_noindex or is_seo_noindex_cat %}checked="checked"{% endif %}
+                                {% if is_seo_noindex_cat %}disabled{% endif %}
```
Comment (robvandenbogaard) [inline modules/mod_admin/templates/_admin_edit_content_seo.tpl:101]: Should this be merged with the condition at line 99? The duplication probably doesn't matter that much actually, in practice.

File: modules/mod_base/controllers/controller_file.erl (hunk 2 above)
Comment (mworrell) [reply modules/mod_base/controllers/controller_file.erl:150]: I think so, yes.

File: modules/mod_seo_sitemap/models/m_seo_sitemap.erl (hunk 3 above)
Comment (mworrell) [reply modules/mod_seo_sitemap/models/m_seo_sitemap.erl:171]: Weird that this seemed to work, and compiled...

File: modules/mod_admin/templates/_admin_edit_content_seo.tpl (hunk 4 above)
Comment (mworrell) [reply modules/mod_admin/templates/_admin_edit_content_seo.tpl:101]: Ah good one, indeed better to merge.

**Pull request merged**